jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -184,6 +174,48 @@ public void onClick(View view) {
         });
     }
 
+    /** Sends the user entered message if permission was granted. Requests the permission if the app
+     * does not already have it. Also shows a dialog if now message was entered. */
+    private void sendMessage() {
+
+        message = editMessage.getText().toString();
+
+        if (message.length() <= 0) {
+            EnterMessageDialogFragment dialog = new EnterMessageDialogFragment();
+            dialog.show(getFragmentManager(), ENTER_MESSAGE_FRAGMENT_TAG);
+            return;
+        }
+
+        delegate.sendMessage(message);
+        editMessage.setText("");
+
+        if (phoneNumberHasNotBeenEntered()) {
```

Comment:
Is this to be preferred over `!phoneNumberHasBeenEntered()`, or is the latter better? Or maybe it doesn't matter?